Reformat Code Files

alexisreytech opened the pull request:
Fixes #92 by using the [Prettier code extension](https://prettier.io/) to reformat the code files for the main website. @brian-montgomery, Prettier is definitely opinionated, so let me know what you think; but I like that this helps create some more consistency across files. There is also an option to have the files be formatted for each commit, but I did not do that as I was not sure if that is something you would want or not. Information for that is in [this section on Git hooks from the Prettier docs](https://prettier.io/docs/en/install#git-hooks).

brian-montgomery reviewed (approved):
I took a look. Prettier feels a lot like [black](https://black.readthedocs.io/en/stable/index.html) (used in Python) to me. It creates an opinionated formatting structure that is easier to parse for a machine, but takes a while for a human to get used to it.

I agree that having an automated syntax standard is a good idea. The more people work on the codebase the more the differing styles become an issue. This seems to be pretty popular, so let’s give it a try and see if we get used to it.

I added a few things to your branch:
* Added a couple of config options. Prettier added back the end spaces in `ExternalLink` among others, so I flipped a couple of switches and added an ignore or two.
* Added the `prettier-plugin-astro` to help it better understand `.astro` files.
* Added some additional files to the ignore list.
* Added a command to run `prettier` in the `Makefile`.

Let me know what you think about my updates.

I'm still trying to get a GitHub workflow to verify the `prettier` output on check in, but I haven't figured out how to execute it correctly so far.

alexisreytech reviewed (commented):
Looks good to me. Thanks for making those changes.

alexisreytech commented:
> I took a look. Prettier feels a lot like [black](https://black.readthedocs.io/en/stable/index.html) (used in Python) to me. It creates an opinionated formatting structure that is easier to parse for a machine, but takes a while for a human to get used to it.
> 
> I agree that having an automated syntax standard is a good idea. The more people work on the codebase the more the differing styles become an issue. This seems to be pretty popular, so let’s give it a try and see if we get used to it.
> 
> I added a few things to your branch:
> 
>     * Added a couple of config options. Prettier added back the end spaces in `ExternalLink` among others, so I flipped a couple of switches and added an ignore or two.
> 
>     * Added the `prettier-plugin-astro` to help it better understand `.astro` files.
> 
>     * Added some additional files to the ignore list.
> 
>     * Added a command to run `prettier` in the `Makefile`.
> 
> 
> Let me know what you think about my updates.
> 
> I'm still trying to get a GitHub workflow to verify the `prettier` output on check in, but I haven't figured out how to execute it correctly so far.

Looks good to me. Thanks for making those changes.
